### DISCUSSION TITLE
Successfully flashed micropython doesn'r run

### DISCUSSION THREAD

**Discussion by bristol406** [micropython/micropython#12953] [ESP32]
I am trying to flash micropython onto an ESP32R4 relay controller module. The esp32 chip is labeled as ESP32 WROVER-B. The flashing process 
completes successfully

----------------------------------------------------------------------------------------------------------------------------------------------------------
E:\MicroPython>esptool.py --port com3 --baud 115200 write_flash --flash_size=detect 0 e:\Micropython\esp32-20220618-v1.19.1.bin
esptool.py v3.1
Serial port com3
Connecting...
Device PID identification is only supported on COM and /dev/ serial ports.
..
Detecting chip type... ESP32
Chip is ESP32-D0WD (revision 1)
Features: WiFi, BT, Dual Core, 240MHz, VRef calibration in efuse, Coding Scheme None
Crystal is 40MHz
MAC: 4c:75:25:48:b5:00
Uploading stub...
Running stub...
Stub running...
Configuring flash size...
Auto-detected Flash size: 16MB
Flash will be erased from 0x00000000 to 0x0017dfff...
Compressed 1560976 bytes to 1029132...
Wrote 1560976 bytes (1029132 compressed) at 0x00000000 in 90.9 seconds (effective 137.3 kbit/s)...
Hash of data verified.

Leaving...
Hard resetting via RTS pin...

E:\MicroPython>
---------------------------------------------------------------------------------------------------------------------------------------------------------

but when I reboot the board I get the error:

rst:0x10 (RTCWDT_RTC_RESET),boot:0x13 (SPI_FAST_FLASH_BOOT)
flash read err, 1000
ets_main.c 371
ets Jun  8 2016 00:22:57

I have tried the following versions of micropython firmware from micropython.org but get the same effect:

esp32-20220618-v1.19.1.bin

ESP32_GENERIC-20230426-v1.20.0.bin

I have tried erasing the flash before flashing micropython but that made no difference. As a test I flashed Tasmota firmware to the board and that did run successfully.

Any guidance would be appreciated.

**Comment by robert-hh** [comment]
You can ignore the `flash read err`. That is common to all WROVER devices. You can use the firmware versions with SPIRAM, but the ones without SPIRAM should work as well. Which terminal emulator do you use?

**Comment by bristol406** [comment]
I'm using Putty terminal emulator which has worked on previous projects. I can try another if you have a suggestion.

**Reply by bristol406** [reply]
That's on Windows 10 PC

**Comment by glenn20** [comment]
> E:\MicroPython>esptool.py --port com3 --baud 115200 write_flash --flash_size=detect 0 e:\Micropython\esp32-20220618-v1.19.1.bin 

try this instead:

```
esptool.py --port com3 --baud 115200 --flash_size=detect write_flash -z 0x1000 e:\Micropython\esp32-20220618-v1.19.1.bin 
```

The firmware files for orig esp32 devices should be written to an offset of `0x1000`. The `-z` option says to use compression so ir flashes a bit quicker (see https://micropython.org/download/ESP32_GENERIC/). You should also be able to use a faster baud rate.

**Reply by bristol406** [reply]
Yes, many thanks, that worked. I had to erase the flash first - probably a good idea anyway. You've saved me some hours of head scratching, thanks again.
